evap/student/tests/test_views.py

```diff
@@ -559,3 +561,109 @@ def test_xmin_of_all_answers_is_updated(self):
+            Evaluation, state=Evaluation.State.IN_EVALUATION, participants=[cls.user, cls.user2]
+        )
+
+        cls.evaluation.general_contribution.questionnaires.add(cls.dropout_questionnaire, cls.normal_questionnaire)
+
+    def assert_no_answer_set(self, form, dropout_questionnaire: Questionnaire):
+        for name, fields in form.fields.items():
+            if name is not None and name.startswith("question_"):
+                _, questionnaire_id, _, _ = parse_answer_field_id(name)
+                field = fields[0]
+
+                if field.tag == "textarea":
+                    self.assertEqual(
+                        field.value,
+                        "",
+                        "Answers to textarea-Questions in the general contribution should be empty",
+                    )
+                    continue
+
+                if questionnaire_id == dropout_questionnaire.id:
+                    self.assertIsNone(field.value, "dropout questionnaires should not be preselected")
+                    continue
+
+                self.assertEqual(
+                    field.value,
+                    str(NO_ANSWER),
+                    "Answers to Questions in the general contribution should be set to NO_ANSWER",
+                )
+
+    def test_choosing_dropout_sets_to_no_answer(self):
+        response = self.app.get(url=reverse("student:drop", args=[self.evaluation.id]), user=self.user, status=200)
+        form = response.forms["student-vote-form"]
+
+        self.assertIn(
+            answer_field_id(self.evaluation.general_contribution, self.dropout_questionnaire, self.question),
+            form.fields.keys(),
+            "The dropout Questionnaire should be shown",
+        )
+
+        self.assert_no_answer_set(form, dropout_questionnaire=self.dropout_questionnaire)
+
+    def test_dropout_possible_iff_dropout_questionnaire_attached(self):
+        self.assertTrue(self.evaluation.is_dropout_allowed)
+        self.assertTrue(
+            self.evaluation.general_contribution.questionnaires.filter(type=Questionnaire.Type.DROPOUT).exists()
+        )
+
+        normal_questionnaires = self.evaluation.general_contribution.questionnaires.exclude(
+            type=Questionnaire.Type.DROPOUT
+        ).all()
+        self.evaluation.general_contribution.questionnaires.set(normal_questionnaires)
+
+        self.assertFalse(self.evaluation.is_dropout_allowed)
+        self.assertFalse(
+            self.evaluation.general_contribution.questionnaires.filter(type=Questionnaire.Type.DROPOUT).exists()
+        )
+
+    def test_view_not_shown_if_dropout_not_allowed(self):
+        normal_questionnaires = self.evaluation.general_contribution.questionnaires.exclude(
+            type=Questionnaire.Type.DROPOUT
+        ).all()
+        self.evaluation.general_contribution.questionnaires.set(normal_questionnaires)
+
+        _ = self.app.get(url=reverse("student:drop", args=[self.evaluation.id]), user=self.user, status=400)
```

Comment:
`test_view_not_shown` makes me expect that the test asserts that something isn't on the page, but there is no assertion here, just the 400 status. Can we maybe change the name?

Maybe also remove the `_ =`, we don't do that anywhere else in the codebase